### DISCUSSION TITLE
Create an Admin Role?

### DISCUSSION THREAD

**Discussion by jvoiptel** [0perationPrivacy/VoIP#92] [Ideas]
I'm wondering what you guys think of this approach. Currently, there's a limitation with Telnyx's API implementation. Their API keys allow you to pretty much do anything, including access all DIDs.  Given their API model, it requires each profile account in this app to have it's own Telnyx account. I think this is limiting for those of us that manage multiple DIDs with a single Telnyx account where DIDs are owned by other users. 

I'm wondering if there could be some kind of admin role where an admin can create profile accounts for users , add the API key and assign the DID. However, when the user signs on, they cannot see the API key. They cannot change their DID. They would not be admins, just users. Does this make sense?

**Comment by fightforprivacy** [comment]
This is a phenomenal idea. 

Can you create a file/description of what each role would have? 

Then work can begin.

**Comment by jvoiptel** [comment]
## VOIP Suite User Roles

| Role | Email Settings| Profile Settings | Account Settings| MFA Settings|Add New Profile|Manage Connections|
| --- | :---: | :---:|---|:---:|:---:|:---:|
| Admin |  :heavy_check_mark: | :heavy_check_mark:|  [x] Change Username <br> [x] Change Password <br> [x] Delete Account <br> [x] Fallback Setting| :heavy_check_mark:| :heavy_check_mark:|
| User Manager | :heavy_check_mark: | |[  ] Change Username <br> [x] Change Password <br> [  ] Delete Account <br> [  ] Fallback Setting | :heavy_check_mark:||:heavy_check_mark:||
| User | :heavy_check_mark: | |[  ] Change Username <br> [x] Change Password <br> [  ] Delete Account <br> [  ] Fallback Setting | :heavy_check_mark:||||

This is a start. There are some things that need to be figured out: 

- Admin should see all DIDs with API key.
- Admin should assign one or more DIDs per profile created.
- Admin only account that can assign roles.
- New concept of Connections where a connection is linked to a single DID.

Also, currently all profile accounts share contacts. Not sure how that will work with this model. Hope this helps.

**Reply by 0perationPrivacy** [reply]
I get what you're saying, but this app is meant to be for 1 individual only with multiple profiles. An "account" is not meant to have multi-tenancy, only multiple aliases. The entire database structure would need to change to add "roles" to users for the "same" account. It would break the login and profile process as you would need to start fresh in the db.

If and when twilio fixes the permissions and roles issue in their API, this problem will be solved as you would be able to issue multiple keys per DID with specific permissions.

Great idea 👍🏻 for someone managing multiple clients, but the db architecture would have to be re-written along with many functions.
(The privacy dashboard is actually written this way and it required a lot of thought and time in the design stage.)
